qln/msghandler.go

```diff
@@ -53,6 +47,28 @@ func (nd *LitNode) PeerHandler(msg lnutil.LitMsg, q *Qchan, peer *RemotePeer) er
 
 }
 
+// BroadcastMessage sends a chat message to all connected websocket clients
+func (nd *LitNode) BroadcastChatMessage(msg lnutil.LitMsg) error {
+	chat, ok := msg.(lnutil.ChatMsg)
+	if !ok {
+		return fmt.Errorf("can't cast to chat message")
+	}
+
+	// get the current set of channels and save them, they may be deleted by litrpc
+	// before the iteration is over
+	var channels []chan lnutil.ChatMsg
```

Comment:
The chan is never closed, after it's used I'm expecting golang garbage collection to take care of it But each chan is only ever used once, when a peer wants a new message they make a new chan.

and it's expected for nd.UserWsCons to change after iterating over it, it's supposed to change as in litrpc clients are removing a reference to the old channel and creating a reference to a new channel in nd.UserWsCons.